internal/cmp/loadbalancer.go

```diff
@@ -37,60 +37,51 @@ func (lb *loadBalancer) Read(ctx context.Context, d *utils.Data, meta interface{
 }
 
 func (lb *loadBalancer) Update(ctx context.Context, d *utils.Data, meta interface{}) error {
-	return nil
-}
-
-func (lb *loadBalancer) Create(ctx context.Context, d *utils.Data, meta interface{}) error {
-	setMeta(meta, lb.lbClient.Client)
-	createReq := models.CreateLoadBalancerRequest{
-		NetworkLoadBalancer: models.CreateNetworkLoadBalancerRequest{
-			Name:        d.GetString("name"),
-			Description: d.GetString("description"),
-			Enabled:     d.GetBool("enabled"),
-			Visibility:  d.GetString("visibility"),
-			ResourcePermissions: models.EnableResourcePermissions{
-				All: d.GetBool("all"),
-			},
-			Config: models.CreateConfig{
-				AdminState: d.GetBool("admin_state"),
-				Loglevel:   d.GetString("loglevel"),
-				Size:       d.GetString("size"),
-				Tier1:      d.GetString("tier1"),
-			},
-		},
+	id := d.GetID()
+	var updateReq models.CreateLoadBalancerRequest
+	if err := tftags.Get(d, &updateReq.NetworkLoadBalancer); err != nil {
+		return err
 	}
 
-	// Get load balancer type id for NSX-T
-	typeRetry := utils.CustomRetry{}
-	typeRetry.RetryParallel(ctx, meta, func(ctx context.Context) (interface{}, error) {
-		return lb.lbClient.GetLoadBalancerTypes(ctx, map[string]string{
-			nameKey: nsxt,
-		})
+	retry := &utils.CustomRetry{
+		InitialDelay: time.Second * 15,
+		RetryDelay:   time.Second * 30,
+	}
+	_, err := retry.Retry(ctx, meta, func(ctx context.Context) (interface{}, error) {
+		return lb.lbClient.UpdateLoadBalancer(ctx, id, updateReq)
 	})
-
-	typeResp, err := typeRetry.Wait()
 	if err != nil {
 		return err
 	}
 
-	types := typeResp.(models.GetLoadBalancerTypes)
+	return tftags.Set(d, updateReq.NetworkLoadBalancer)
+}
+
+func (lb *loadBalancer) Create(ctx context.Context, d *utils.Data, meta interface{}) error {
+	setMeta(meta, lb.lbClient.Client)
+	var createReq models.CreateLoadBalancerRequest
+	if err := tftags.Get(d, &createReq.NetworkLoadBalancer); err != nil {
+		return err
+	}
+
+	allTypes, _ := lb.lbClient.GetLoadBalancerTypes(ctx, map[string]string{
+		nameKey: nsxt,
+	})
 
-	for i, n := range types.LoadBalancerTypes {
+	for i, n := range allTypes.LoadBalancerTypes {
 		if n.Name == nsxt {
-			createReq.NetworkLoadBalancer.Type = types.LoadBalancerTypes[i].Name
-			createReq.NetworkLoadBalancer.NetworkServerID = 1
+			createReq.NetworkLoadBalancer.Type = allTypes.LoadBalancerTypes[i].Name
+			createReq.NetworkLoadBalancer.NetworkServerID = networkServerID
```

Comment:
Can you explain this ? I hope u are getting the LB Type here. refer tier0 gateway type or nsx network type.